### ISSUE TITLE
A question about the RL training

### ISSUE THREAD

**Issue by FayeXXX** [seq-to-mind/semi-style-transfer#4]
Hi, when I set the RL_training to True, I found that in your paper you apply both greedy decoding generation and sampling generation as the reconstruction reward, and then I run line262 instead of 265:
https://github.com/seq-to-mind/semi-style-transfer/blob/79ea10ba042967afbed426d76684e06122ae994d/Model.py#L262-L265

And then I got the  GAN_generator_loss which is around -0.025 and the cyclic loss is negative either.  I have no idea how it works when the loss is a negative number instead of a positive number.

**Comment by seq-to-mind** [comment]
The self-critical RL advantage might be < 0 when the reward value obtained by sampling generation is lower than greedy decoding generation. For such data samples, the RL will add penalty to the sampling generation.

**Comment by seq-to-mind** [comment]
Moreover, if you do not wanna such RL negative advantage, you can add a clamp to the advantage value; for example: add `cyclic_advantage = torch.clamp(cyclic_advantage, min=0.0)` after Line 265.

**Comment by FayeXXX** [comment]
Thank you for your reply, and I will wait for the final result. 
And have a good day~

**Comment by FayeXXX** [comment]
BTW, I've noticed that there are 4 references for each test data in Yelp. But in most of other implementations there are only  2 references, and I'm wondering how did you collect these data.

**Comment by seq-to-mind** [comment]
The Yelp data with 4 human references were collected from some previous work, here is one `https://github.com/zhijing-jin/IMaT/tree/master`

**Comment by FayeXXX** [comment]
Thank you so much for your kindness and your quick reply.

**Comment by FayeXXX** [comment]
Hi, hope everything is going well. 
There is a question when I reproduce your code on Amazon dataset. There are 30840 pairs of data in pseudo_paired_data_Amazon/semantic_based, but not enough to gain a good performance as mentioned in your paper. So I run 'find_sentence_Semantic_Match.py' and try to generate more data. But I found there are only 4000 pairs under the default setting :'if similar_scores[one_best] > 0.85'(line 77),  that means most similar scores are lower than 0.85. If I want to generate more data, dose that mean I should change the threshold from 0.85 to 0.6 or lower numbers? But lower number may lead to data pairs for training with poor quality. I have no idea how to gain more training data.
 Any advice from you  will be really appreciated. Thank you.

**Comment by seq-to-mind** [comment]
The data of 30K pairs are sufficient to get reasonable performance. What kind of the result you obtained in your current setting?

If you really want to get more data by running `find_sentence_Semantic_Match.py`, you can adjust the value 'similar_scores[one_best] > 0.85'(line 77) to a lower one.

**Comment by FayeXXX** [comment]
Thank you so much for  your quick reply.
1. The result I obtained in my current setting is that the model generates 'it was worth the money ' repeatedly, then I use larger decay_rate which controls the weight of the content preservation loss ,  and the repetition decreases but still exists. I'm wondering is that because of the data insufficient, since there are 27w training data in the original dataset but only 3w in the pseudo_paired_data_Amazon.
2. As for your advice to the lower value, I'm trying to train.
